invenio_users_resources/services/users/service.py

```diff
@@ -72,6 +77,25 @@ def search(self, identity, params=None, search_preference=None, **kwargs):
             **kwargs,
         )
 
+    def search_all(
```

Comment:
The current service `search` only returns users that are confirmed and active. 

For user moderation, we need to be able to see all users (especially now that we're adding a 'blocking' functionality). So I split the two search. 

`search_all` has its own permission.